packages/dds/tree/src/test/feature-libraries/editable-tree/editableTree.edit.spec.ts

```diff
@@ -0,0 +1,329 @@
+        age: [{ value: 35, type: int32Schema.name }],
+        salary: [{ value: 10420.2, type: float32Schema.name }],
+        friends: [{ fields: {
+            Mat: [{ type: stringSchema.name, value: "Mat" }],
+        }, type: mapStringSchema.name }],
+        address: [{
+            fields: {
+                street: [{ value: "treeStreet", type: stringSchema.name }],
+                phones: [{
+                    type: phonesSchema.name,
+                    fields: {
+                        [EmptyKey]: [
+                            { type: stringSchema.name, value: "+49123456778" },
+                            { type: int32Schema.name, value: 123456879 },
+                            { type: complexPhoneSchema.name, fields: {
+                                number: [{ value: "012345", type: stringSchema.name }],
+                                prefix: [{ value: "0123", type: stringSchema.name }],
+                            } },
+                        ],
+                    },
+                }],
+                simplePhones: [{
+                    type: simplePhonesSchema.name,
+                    fields: {
+                        [EmptyKey]: [
+                            { type: stringSchema.name, value: "112" },
+                        ],
+                    },
+                }],
+                sequencePhones: [
+                    { type: stringSchema.name, value: "113" },
+                    { type: stringSchema.name, value: "114" },
+                ]
+            },
+            type: addressSchema.name,
+        }],
+    },
+};
+
+let _provider: ITestTreeProvider;
+
+async function createSharedTrees(schema: SchemaData, data: JsonableTree, nofTrees = 1): Promise<readonly ISharedTree[]> {
+    const provider = await TestTreeProvider.create(nofTrees + 1);
+    _provider = provider;
+    for (const tree of provider.trees) {
+        assert(tree.isAttached());
+        const forest = tree.forest;
+        forest.schema.updateFieldSchema(rootFieldKey, schema.globalFieldSchema.get(rootFieldKey) ?? fail("oops"));
+        for (const [key, value] of schema.treeSchema) {
+            forest.schema.updateTreeSchema(key, value);
+        }
+    }
+    assert(isEmptyTree(provider.trees[0].root));
+    provider.trees[0].root[insertRootSymbol](singleTextCursor(data));
+    await provider.ensureSynchronized();
+    return provider.trees.slice(1);
+}
+
+describe("editing with editable-tree", () => {
+    describe("Non-sequence fields", () => {
+        it("update property", async () => {
+            const trees = await createSharedTrees(fullSchemaData, personData, 2);
+            const person1 = trees[0].root as PersonType;
+            const person2 = trees[1].root as PersonType;
+            person1.address.street = "bla";
+            assert.equal(person1.address.street, "bla");
+            person1.age = newAge;
+            assert.equal(person1.age, newAge);
+            trees[1].context.prepareForEdit();
```

Comment:
Why are all those `prepareForEdit()` calls necessary (in this test and others)?

Reply:
Thanks for pointing this. Should not be necessary anymore.